cmake/schemes/url_sha1_boost_library.cmake.in

```diff
@@ -235,27 +237,30 @@ file(
     "  : ${toolset_version}\n"
 )
 
-if(MSVC)
+if(WIN32)
   # For Visual Studio C++ flags must not be set in compiler section.
   # Section <compileflags> should be used.
   #   * https://github.com/ruslo/hunter/issues/179
-  file(APPEND ${boost_user_jam} "  : \"${boost_compiler}\"\n")
+  # Even though user.config.jam provides the path to archiver 'b2'
+  # tries to find it based on the path to the compiler. In this 
+  # process if the path contains any spaces or any other invalid 
+  # characters that windows use for its long names, it derives a 
+  # path which is incorrect. To solve this we convert the path to 
+  # 8.3 FileName format or windows short filename format.
+  # For reference check http://stackoverflow.com/a/10227715/2288008
+  hunter_windows_to_snf_filename(${boost_compiler} boost_compiler_snf)
```

Comment:
Yes you are right. It is possible to cross compile MinGW on Linux. I will change it to CMAKE_HOST_WIN32